### PR TITLE
Do not parse table when no fields are defined

### PR DESCRIPTION
**Pull request by jwoillez**
### Description

[astropy/astroquery](https://github.com/astropy/astroquery) can be used to obtain tables from the Vizier service. Some of the answers can correspond to empty (and admittedly non-compliant) tables that look like the following:

    <?xml version="1.0" encoding="utf-8"?>
    
    <VOTABLE version="1.4" xmlns="http://www.ivoa.net/xml/VOTable/v1.4" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="http://www.ivoa.net/xml/VOTable/v1.4">
     <RESOURCE type="results">
      <TABLE>
       <DATA>
        <TABLEDATA>
         <TR></TR>
         <TR></TR>
         <TR></TR>
        </TABLEDATA>
       </DATA>
       <INFO name="matches" value="3">matching records</INFO>
      </TABLE>
     </RESOURCE>
    </VOTABLE>

The current VOTable parser fails in a rather obscure way. This may be worked around by preventing the parsing of `<DATA>` when no `<FIELD>` are given, as suggested in this pull request.

Fixes astropy/astroquery#1691

### CODE REVIEW

**Comment by pllim** [comment]
Thanks! Needs a test and change log.

**Comment by jwoillez** [comment]
@pllim I added both test and change log.

**Comment by bsipocz** [comment]
> astropy-bot:changelog complains. Is that because I added the new entry in the wrong section?

yes, but don't worry about it now, this close to feature freeze we rather merge with the wrong section than rerun CI for the sake of a fixed changelog.

**Comment by jwoillez** [comment]
@tomdonaldson I implemented your suggestion. Should I also look into adding a test for this warning/exception?

**Comment by tomdonaldson** [comment]
> @tomdonaldson I implemented your suggestion. Should I also look into adding a test for this warning/exception?

@jwoillez Thank you for that change, and also for fixing the schema reference!  Yes, adding a test for the new error would be great.  Please let me know if you have any questions about that.

**Comment by jwoillez** [comment]
@tomdonaldson Here it is. Is this the format you were expecting?

**Comment by tomdonaldson** [comment]
The new test seems fine as written.  I'm comfortable that the mechanics of warning vs. errors are being tested elsewhere.

Thanks again for this contribution @jwoillez !

File: astropy/io/votable/tests/table_test.py [hunk 1]
```diff
@@ -186,6 +186,18 @@ def test_empty_table():
...
+def test_no_field_not_empty_table():
+    votable = parse(get_pkg_data_filename('data/no_field_not_empty_table.xml'))
+    table = votable.get_first_table()
+    assert len(table.fields) == 0
+    assert len(table.infos) == 1
+
+
+@raises(E25)
+def test_no_field_not_empty_table_exception():
+    votable = parse(get_pkg_data_filename('data/no_field_not_empty_table.xml'), verify='exception')
```
Comment (pllim) [inline astropy/io/votable/tests/table_test.py:198]: It's a bit cleaner to use `pytest` here. `@raises` was written a long time ago before `pytest` has it.
```suggestion
def test_no_field_not_empty_table_exception():
    with pytest.raises(E25):
        votable = parse(get_pkg_data_filename('data/no_field_not_empty_table.xml'), verify='exception')
```

**Comment by bsipocz** [comment]
Thanks @jwoillez!

**Pull request merged**